packages/common/src/compiler/filtersCompiler.ts

```diff
@@ -144,6 +144,14 @@ export const renderNumberFilterSql = (
             return `(${dimensionSql}) < (${filter.values?.[0] || 0})`;
         case FilterOperator.LESS_THAN_OR_EQUAL:
             return `(${dimensionSql}) <= (${filter.values?.[0] || 0})`;
+        case FilterOperator.IN_BETWEEN:
+            return `(${dimensionSql}) >= (${
+                filter.values?.[0] || 0
+            }) AND (${dimensionSql}) <= (${filter.values?.[1] || 0})`;
+        case FilterOperator.NOT_IN_BETWEEN:
+            return `(${dimensionSql}) <= (${
+                filter.values?.[0] || 0
+            }) OR (${dimensionSql}) >= (${filter.values?.[1] || 0})`;
```

Comment:
@TuringLovesDeathMetal Can you confirm this logic pls
We have implemented the between to be inclusive (same as the date between). Based on that, the "not between" should be `< min OR > max`, so the numbers mentioned are not part of the results. 
I belive `<= min OR >= max` is a bug. 

Example between:
<img width="1649" alt="Screenshot 2025-02-24 at 14 22 08" src="https://github.com/user-attachments/assets/efde5f90-94bb-4ae2-a7a9-4a041ed430c4" /

Example not between:
<img width="1643" alt="Screenshot 2025-02-24 at 14 21 51" src="https://github.com/user-attachments/assets/529c5671-bd93-41f5-a2f7-3e0501eb38aa" />

Reply:
agreed! I think it should be "exclusive" of the numbers listed (so what you said: < min or > max)

Reply:
Makes sense, making the change